src/crimson/net/ProtocolV2.cc

```diff
@@ -0,0 +1,1603 @@
+      }
+    }
+  });
+}
+
+void ProtocolV2::execute_connecting()
+{
+  trigger_state(state_t::CONNECTING, write_state_t::delay, true);
+  seastar::with_gate(pending_dispatch, [this] {
+      enable_recording();
+      global_seq = messenger.get_global_seq();
+      return Socket::connect(conn.peer_addr)
+        .then([this](SocketFRef sock) {
+          socket = std::move(sock);
+          if (state == state_t::CLOSING) {
+            return socket->close().then([this] {
+              logger().info("{} is closed during Socket::connect()", conn);
+              abort_in_fault();
+            });
+          }
+          return seastar::now();
+        }).then([this] {
+          return banner_exchange();
+        }).then([this] (entity_type_t _peer_type,
+                        entity_addr_t _peer_addr) {
+          if (conn.peer_type != _peer_type) {
+            logger().debug("{} connection peer type does not match what peer advertises {} != {}",
+                           conn, conn.peer_type, (int)_peer_type);
+            dispatch_reset();
+            abort_in_close();
+          }
+        }).then([this] {
+          return client_auth();
+        }).then([this] {
+          if (!server_cookie) {
+            ceph_assert(connect_seq == 0);
+            return client_connect();
+          } else {
+            ceph_assert(connect_seq > 0);
+            // TODO: lossless policy
+            ceph_assert(false);
+            return client_reconnect();
+          }
+        }).then([this] (bool proceed_or_wait) {
+          if (proceed_or_wait) {
+            execute_ready();
+          } else {
+            execute_wait();
+          }
+        }).handle_exception([this] (std::exception_ptr eptr) {
+          // TODO: handle fault in CONNECTING state
+          return fault();
+        });
+    });
+}
+
+// ACCEPTING state
+
+seastar::future<> ProtocolV2::_auth_bad_method(int r)
+{
+  // _auth_bad_method() logic
+  ceph_assert(r < 0);
+  std::vector<uint32_t> allowed_methods;
+  std::vector<uint32_t> allowed_modes;
+  messenger.auth_server->get_supported_auth_methods(
+      conn.get_peer_type(), &allowed_methods, &allowed_modes);
+  logger().warn("{} send AuthBadMethod(auth_method={}, r={}, "
+                "allowed_methods={}, allowed_modes={})",
+                conn, auth_meta->auth_method, cpp_strerror(r),
+                allowed_methods, allowed_modes);
+  auto bad_method = AuthBadMethodFrame::Encode(
+      auth_meta->auth_method, r, allowed_methods, allowed_modes);
+  return write_frame(bad_method)
+  .then([this] {
+    return server_auth();
+  });
+}
+
+seastar::future<> ProtocolV2::_handle_auth_request(bufferlist& auth_payload, bool more)
+{
+  // _handle_auth_request() logic
+  ceph_assert(messenger.auth_server);
+  bufferlist reply;
+  int r = messenger.auth_server->handle_auth_request(
+      conn.shared_from_this(), auth_meta,
+      more, auth_meta->auth_method, auth_payload,
+      &reply);
+  if (r == 1) {
```

Comment:
i'd suggest use switch instead of if for choices. also, might want to comment on the "1" and "0" cases.

Reply:
fixed